src/test/java/org/apache/maven/shared/utils/ExpandTest.java

```diff
@@ -326,11 +326,11 @@ private void verifyExpandedFileAndContent( File targetDir, String expectedConten
 
         assertNotNull( expandedFile );
 
-        java.util.Scanner scanner = new java.util.Scanner( expandedFile ).useDelimiter( "\n" );
-        String text = scanner.next();
-
-        assertThat( "expanded file content must match"
-                  , text
-                  , is( expectedContent ) );
+        try (Scanner base = new Scanner( expandedFile );
+             Scanner scanner = base.useDelimiter( "\n" ) ) 
```

Comment:
Having two local variables with the same reference is not necessary. It is the flow design:
https://docs.oracle.com/javase/7/docs/api/java/util/Scanner.html#useDelimiter(java.util.regex.Pattern)

Reply:
This was an interesting corner case I'd not encountered before and it took a while to figure out. It cures a warning about failure to close the first scanner in try-with-resources block. Though looking at it now I see another way to approach this.